src/engine/sys/sdl_glimp.cpp

```diff
@@ -962,67 +962,55 @@ GLimp_InitExtensions
 ===============
 */
 
-static void RequireExt( bool hasExt, const char* name )
-{
-	if ( hasExt )
-	{
-		logger.WithoutSuppression().Notice("...using GL_%s", name );
-	}
-	else
-	{
-		Sys::Error("Required extension GL_%s is missing", name );
-	}
-}
+#define NoticeExtUnsupported(name) logger.WithoutSuppression().Notice("...GL_%s unsupported", #name)
 
-static bool LoadExtWithCvar( bool hasExt, const char* name, bool cvarValue )
+static bool LoadExt(bool isRequired, bool isCore, bool hasExt, const char* name, bool cvarValue)
 {
-	if ( hasExt )
+	if (hasExt || (isCore && glConfig2.glCoreProfile))
 	{
-		if ( cvarValue )
+		if (cvarValue)
 		{
-			logger.WithoutSuppression().Notice("...using GL_%s", name );
+			logger.WithoutSuppression().Notice("...using GL_%s", name);
 			return true;
 		}
+		else if (isRequired)
+		{
+			DAEMON_ASSERT("Required extension can't be made optional");
```

Comment:
asserting a value which is always true

Reply:
`isRequired` is `true` with `LOAD_REQUIRED_EXTENSION` and `LOAD_REQUIRED_CORE_EXTENSION` but `false` with others.

Reply:
The assert macro terminates the program when it was built in debug mode and the argument evaluates to false as a boolean. A string literal is always non-null, meaning it is true as a boolean, so nothing happens.

Reply:
Ah, I see what you mean. I guess I wanted to call `Sys::Error()` instead :woman_facepalming:

Reply:
Ah no, this error would be a coding bug, so ASSET would be a good idea anyway, I just have to redo it better.

Anyway, why this file also has an `ASSERT` somewhere else? Are they the same?
https://github.com/DaemonEngine/Daemon/blob/6b94e2e890f893afdc55b7f70bd58cb39f3cafcd/src/engine/sys/sdl_glimp.cpp#L263-L265

Reply:
The condition `smpData == nullptr` could be either true or false, depending on the value of `smpData`. So it makes sense.

Reply:
Sorry, my question was: is `DAEMON_ASSERT` the same as `ASSERT`?

Reply:
`DAEMON_ASSERT` is the same as `ASSERT`